lib/NormalModule.js

```diff
@@ -136,17 +135,32 @@ class NormalModule extends Module {
 	}
 
 	createLoaderContext(resolver, options, compilation, fs) {
+		const addErrorFrom = (error) => {
+			if(!error.from) {
+				const currentLoaderObject = loaderContext.loaders && loaderContext.loaders.length ?
+					loaderContext.loaders[loaderContext.loaderIndex] :
+					null;
+				if(currentLoaderObject) {
+					const loaderName = currentLoaderObject.origin || currentLoaderObject.path;
+					error.from = /\//.test(loaderName) ?
+						path.relative(loaderContext.rootContext, loaderName) :
+						loaderName;
+				}
+			}
+		};
 		const loaderContext = {
 			version: 2,
 			emitWarning: (warning) => {
 				if(!(warning instanceof Error))
 					warning = new NonErrorEmittedError(warning);
+				addErrorFrom(warning);
 				this.warnings.push(new ModuleWarning(this, warning));
 			},
 			emitError: (error) => {
 				if(!(error instanceof Error))
 					error = new NonErrorEmittedError(error);
-				this.errors.push(new ModuleError(this, error));
+				addErrorFrom(error);
+				this.errors.push(new ModuleBuildError(this, error));
```

Comment:
ModuleBuildError -> ModuleError

and improve the message in `ModuleError`.

These are two different kind of errors.